src/AzureAppConfigurationImpl.ts

```diff
@@ -291,17 +417,73 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
                 this.#refreshTimer.backoff();
                 throw error;
             }
+            return Promise.resolve(true);
+        }
 
-            // successfully refreshed, run callbacks in async
-            for (const listener of this.#onRefreshListeners) {
-                listener();
+        return Promise.resolve(false);
+    }
+
+    /**
+     * Refresh feature flags.
+     * @returns true if feature flags are refreshed, false otherwise.
+     */
+    async #refreshFeatureFlags(): Promise<boolean> {
+        // if still within refresh interval/backoff, return
+        if (!this.#featureFlagRefreshTimer.canRefresh()) {
+            return Promise.resolve(false);
+        }
+
+        // check if any feature flag is changed
+        let needRefresh = false;
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter,
+                pageEtags: selector.pageEtags
+            };
+            const pageIterator = listConfigurationSettingsWithTrace(
+                this.#requestTraceOptions,
+                this.#client,
+                listOptions
+            ).byPage();
+
+            let pageCount = 0;
+            for await (const page of pageIterator) {
+                if (page._response.status === 200) { // created or changed
+                    needRefresh = true;
+                    break;
+                }
+                // unchanged, check next page
+                pageCount++;
+            }
+
+            if (pageCount !== selector.pageEtags?.length) {
+                needRefresh = true; // page count changed indicating feature flags are added or deleted
+            }

```

Comment:
This check is not needed. Service ensures that if pages are added or deleted, page etag will change, leading to 200 response. Thus, we don't need to maintain page count on client side.

Reply:
A sample: 

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/yanzh/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/yanzh/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"Aptos Narrow", sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{white-space:normal;}
-->
</style>
</head>

<body link="#467886" vlink="#96607D">


items | pages | etags
-- | -- | --
99 | 1 | vl5Cd2p4D1lCoNLt7hZhoC2i6fYNCbRQu9vMWI2-Hxw
100 | 2 | Rgl_lrYN54YE5ABHRondZJQMvTKtISxZtxiCewDkK30     00wWELzclWef22f82qcUDI7FbJweVOoKWZDA_JoUeD0
101 | 2 | Rgl_lrYN54YE5ABHRondZJQMvTKtISxZtxiCewDkK30     HYBJUIuXa4rJRH7dfYomEEz_RLiEpFqxhn81I3_aW38



</body>

</html>

Reply:
> Service ensures that if pages are added or deleted, page etag will change,

Indeed. I'll remove the pageCount check.

Reply:
fb56a0f